pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -308,3 +334,62 @@ func createValidImageVersionLabel(image string) string {
 	}
 	return cephImageVersion
 }
+
+func waitForPvcToExpandWithTimeout(ctx context.Context, client client.Client, pvcResizeMap map[string]pvcResize, namespace string, pvcResizeWaitTimeOut time.Duration) {
+	currentTime := time.Now()
+	if pvcResizeWaitTimeOut == 0 {
+		pvcResizeWaitTimeOut = defaultWaitTimeoutForHealthyOSD
+	}
+	timeout := currentTime.Add(pvcResizeWaitTimeOut)
+	allPvcResize := false
+	for !allPvcResize {
+		if time.Now().After(timeout) {
+			logger.Errorf("timed out waiting for PVC size to be updated, osds may need to be restarted manually after resize")
+			return
+		}
+		allPvcResize = checkAllPvcResize(ctx, client, namespace, pvcResizeMap)
+		if !allPvcResize {
```

Comment:
If changing the for loop, then would need this...
```suggestion
		if allPvcResize {
			logger.Info("all osd pvcs resized")
			return
		}
		// wait for 5 seconds before checking again
		time.Sleep(5 * time.Second)
```